kolibri/plugins/registry.py

```diff
@@ -48,14 +67,15 @@ def initialize():
             try:
                 plugin_module = importlib.import_module(app + ".kolibri_plugin")
                 logger.debug("Loaded kolibri plugin: {}".format(app))
+                all_classes = dict([(name, cls) for name, cls in plugin_module.__dict__.items() if isinstance(cls, type)])
                 plugin_classes = []
-                for obj in plugin_module.PLUGINS:
+                for obj in all_classes.values():
                     if type(obj) == type and issubclass(obj, KolibriPluginBase):
                         plugin_classes.append(obj)
                 for plugin_klass in plugin_classes:
-                    plugin_obj = plugin_klass()
-                    for hook, callback in plugin_obj.hooks().items():
-                        hooks.register_hook(hook, callback)
+                    # Initialize the class, nothing more happens for now.
+                    logger.debug("Initializing plugin: {}".format(plugin_klass.__name__))
+                    plugin_klass()
```

Comment:
Why instantiate a plugin class at all if you don't keep a reference to that object? I think in this case I would prefer that we don't expose plugin instantiation as a sort of "on hook registration" event but instead have each hook class explicitly define a class method. I do not find it very obvious to have `__init__` serve this purpose, and I think it creates also the possibility that a hook class author will mistakenly instantiate a hook class at some other point in the life cycle, not realizing that it should only be done here.